packages/platform-android/src/commands/runAndroid/index.ts

```diff
@@ -198,88 +194,6 @@ function runOnSpecificDevice(
   }
 }
 
-function buildApk(gradlew: string, sourceDir: string) {
-  try {
-    // using '-x lint' in order to ignore linting errors while building the apk
-    const gradleArgs = ['build', '-x', 'lint'];
-    logger.info('Building the app...');
-    logger.debug(`Running command "${gradlew} ${gradleArgs.join(' ')}"`);
-    execa.sync(gradlew, gradleArgs, {stdio: 'inherit', cwd: sourceDir});
-  } catch (error) {
-    throw new CLIError('Failed to build the app.', error);
-  }
-}
-
-function tryInstallAppOnDevice(
-  args: Flags,
-  adbPath: string,
-  device: string,
-  androidProject: AndroidProject,
-) {
-  try {
-    // "app" is usually the default value for Android apps with only 1 app
-    const {appName, sourceDir} = androidProject;
-    const {appFolder} = args;
-    const variant = args.variant.toLowerCase();
-    const buildDirectory = `${sourceDir}/${appName}/build/outputs/apk/${variant}`;
-    const apkFile = getInstallApkName(
-      appFolder || appName, // TODO: remove appFolder
-      adbPath,
-      variant,
-      device,
-      buildDirectory,
-    );
-
-    const pathToApk = `${buildDirectory}/${apkFile}`;
-    const adbArgs = ['-s', device, 'install', '-r', '-d', pathToApk];
-    logger.info(`Installing the app on the device "${device}"...`);
-    logger.debug(
-      `Running command "cd android && adb -s ${device} install -r -d ${pathToApk}"`,
-    );
-    execa.sync(adbPath, adbArgs, {stdio: 'inherit'});
-  } catch (error) {
-    throw new CLIError('Failed to install the app on the device.', error);
-  }
-}
-
-function getInstallApkName(
-  appName: string,
-  adbPath: string,
-  variant: string,
-  device: string,
-  buildDirectory: string,
-) {
-  const availableCPUs = adb.getAvailableCPUs(adbPath, device);
-
-  // check if there is an apk file like app-armeabi-v7a-debug.apk
-  for (const availableCPU of availableCPUs.concat('universal')) {
-    const apkName = `${appName}-${availableCPU}-${variant}.apk`;
-    if (fs.existsSync(`${buildDirectory}/${apkName}`)) {
-      return apkName;
-    }
-  }
-
-  // check if there is a default file like app-debug.apk
-  const apkName = `${appName}-${variant}.apk`;
-  if (fs.existsSync(`${buildDirectory}/${apkName}`)) {
-    return apkName;
-  }
-
-  throw new CLIError('Could not find the correct install APK file.');
```

Comment:
I am wondering if we could somehow handle #124 here?

Reply:
Also, these tests might be useful https://github.com/react-native-community/cli/pull/934/files

Reply:
Is #124 still an issue?

Reply:
@Krizzu wanted to follow up on your question and state that #124 does appear to be occurring for me. I have [captured my details on the issue](https://github.com/react-native-community/cli/issues/124#issuecomment-611133214).

Reply:
@dcalhoun 
I'm addressing your issue in this PR, but to make it correct, we'd have to put some constraints/assumptions first:
- No flavour mixing (multiple flavour)
- flavour name must be lowercase

Otherwise, we'd have to find a better way (either different CLI arguments or something else) than what [I did here](https://github.com/react-native-community/cli/commit/9100b764a132ff17f663af83fa0d01c00141eb97#diff-a1c62b130fddb803e053cba3ec05f980R189) 

cc @thymikee

Reply:
@Krizzu thanks! The second constraint should be easy enough to meet. The first a bit more difficult, as we build our Android app for two different stores (Google Play, Amazon), different environments (Production, Staging), and toggle debuggability with build variants. I suppose there may be a way to duplicate config, rather than composing it, but I'll have to look into it some.

Thank you regardless of the response and efforts.

Reply:
@dcalhoun the constraints I mention are by no means set in stone. Actually, if we only keep the second one, we could actually support flavour mixing.

Reply:
Ah, OK. The casing of build variant names most definitely carries almost no value to me. If enforcing lowercase allows for composing build variants, that would be my preferences.

Reply:
I've added flavour mixing support, FYI @dcalhoun @thymikee